### PR TITLE
Add a script to submit tests to TFT locally

### PR DESCRIPTION
**Pull request by zhukovgreen**
Steps:
- [x] add script which submits the tmt plans to the testing farm
The quality of this script:
0. cli utility in the scripts directory
1. Automatically detect all plans and environments and translate this into the right API calls
2. Track the status of submitted jobs and show the debug links
3. Secrets are managed with the help of dotenv
- [x] unify the solution of managing secrets to just dot env as a single source of it. Ansible will use lookup plugin and .env defined vars should be loaded first (for local run)
- [x] automate the process of getting the right ref of the repo (param in API call)

This MR introduces new cli for interaction with the TFT API from the local development environment:
```bash
➞  python scripts/tft --help                                                          [git:feature/tft-integration] ✖  
Usage: tft [OPTIONS]

  Script to interact with testing farm service.

  Ensure the content of .env.

  Example commands: ``` # Submit only given plan to tft and use existing
  copr rpm builds python scripts/tft -v -p /plans/integration/inhibit-if-
  kmods-is-not-supported/centos8 --copr-build-id 2353320 --copr-build-id
  2353321

  # Run all plans for some-atypical-remote-name git remote with debug mode
  (-v) python scripts/tft -v -r some-atypical-remote-name ```

Options:
  -p, --plans TEXT          Plan names. i.e. -p /plan/name. Could be multiple
                            [default: /plans/]

  -r, --remote-name TEXT    Git remote name from which the content of the repo
                            will be cloned at current commit. Warning: changes
                            should be pushed to the remote before running this
                            script.  [default: origin]

  -b, --copr-build-id TEXT  Use specified builds ids instead of creating a new
                            copr builds.

  -v, --verbose
  --help                    Show this message and exit.
```

To submit the following tmt plans `/plans/integration/inhibit-if-kmods-is-not-supported/centos8` you need to do run:
```bash
$ rm .install
$ make install
$ source .venv3/bin/activate
$ python scripts/tft -v -p /plans/integration/inhibit-if-kmods-is-not-supported/centos8
```
note: first two commands needed to be run just once (to update the local deps)

### CODE REVIEW

**Comment by ZhukovGreen** [comment]
@oamg/convert2rhel-developers pls review

**Comment by bocekm** [comment]
I've done a few tweak for things that were failing.
Now the tests are submitted on my end but end with `Unsupported provision method 'libvirt' in the '/plans/integration/inhibit-if-kmods-is-not-supported/centos8' plan.`

**Comment by bocekm** [comment]
TODO: we can submit just one copr build because copr builds the package for all chroots we need - RHEL 7 and 8. Currently it submits two copr jobs or asks for two copr builds through the `--copr-build-id` option.

File: scripts/tft/app.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+
+assert sys.version_info > (3, 7), "Script can't be run with lower version of python"
+
+
+# Mapping of tmt plans origin_vm_name provisioner definition to tft composes
+VM2COMPOSES = {
+    "c2r_centos7_template": "CentOS-7",
+    "c2r_centos8_template": "CentOS-8",
+    "c2r_oracle7_template": "Oracle-Linux-7.9",
+    "c2r_oracle8_template": "Oracle-Linux-8.4",
```
Comment (bocekm) [inline scripts/tft/app.py:31]:
```suggestion
    "c2r_centos7_template": "CentOS-7-latest",
    "c2r_centos8_template": "CentOS-8-latest",
    "c2r_oracle7_template": "Oracle-Linux-7.9",
    "c2r_oracle8_template": "Oracle-Linux-8.4",
```
The internal TFT (which we use) has different compose names than the public TFT.

File: scripts/tft/models.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+class Environment(BaseModel):
+    arch: CPUArch
+    os: Os
+    variables: Dict[str, SecretStr]
+    artifacts: List[Artifact]
+
+
+class TFTRequest(BaseModel):
+    api_key: SecretStr = SecretStr(env.str("TFT_API_KEY"))
```
Comment (bocekm) [inline scripts/tft/models.py:43]: The TFT API key is leaking to the TFT request and anyone can see it, e.g. https://api.dev.testing-farm.io/v0.1/requests/1abf7ac9-f1e0-490f-b9b7-1cda0e61b2c4. We need to fix that. The one in the example is already removed as it's been leaked.

**Review by bocekm — CHANGES_REQUESTED** [review]

File: scripts/tft/models.py (hunk 2 above)
Comment (ZhukovGreen) [reply scripts/tft/models.py:43]: we can't do better on our side. there is a relevant ticket on tft https://gitlab.com/testing-farm/general/-/issues/27

Comment (bocekm) [reply scripts/tft/models.py:43]: @thrix ^^

**Comment by kokesak** [comment]
We will no further need this feature as most of this work is already done by packit. Submitting jobs to TF will be explained in separate documentation.

**Pull request closed without merging**